### PR TITLE
[v6r15] Data Logging System

### PR DESCRIPTION
**Pull request by coberger**
This is the data logging system for DIRAC. It allows to track all operations on files. It is transparent for users, it does not change the behavior of DIRAC. It is flexible for future additional.

Here is an example of the logging of a put and register operation and of a remove file operation : 
$ python dirac/DIRAC/DataManagementSystem/scripts/dirac-dls-sequence.py /lhcb/sub/file2.data

Sequence 1, Caller dirac-dms-add-file.py , UserName coberger , HostName localhost.localdomain , Group devGroup 
	DataManager.putAndRegister, 2015-08-19 11:50:13, Successful, TargetSE StorageTest1
		DataManager.registerFile, 2015-08-19 11:50:14, Successful, TargetSE StorageTest1
			FileCatalog.addFile, 2015-08-19 11:50:14, Successful, TargetSE StorageTest1
		StorageElement.putFile, 2015-08-19 11:50:14, Successful, TargetSE StorageTest1

Sequence 2, Caller dirac-dms-remove-files.py , UserName coberger , HostName localhost.localdomain , Group devGroup 
	DataManager.removeFile, 2015-08-19 12:38:20, Successful
		FileCatalog.removeFile, 2015-08-19 12:38:20, Successful
		FileCatalog.removeReplica, 2015-08-19 12:38:20, Successful, TargetSE StorageTest1
		StorageElement.removeFile, 2015-08-19 12:38:20, Successful, TargetSE StorageTest1

It is also possible to have all method calls about one LFN , here is the output: 

$ python dirac/DIRAC/DataManagementSystem/scripts/dirac-dls-method-call.py /lhcb/sub/file2.data

Found 8 method calls
2015-08-19 11:50:13, DataManager.putAndRegister, TargetSE StorageTest1, Successful, SequenceID 1
2015-08-19 11:50:14, DataManager.registerFile, TargetSE StorageTest1, Successful, SequenceID 1
2015-08-19 11:50:14, StorageElement.putFile, TargetSE StorageTest1, Successful, SequenceID 1
2015-08-19 11:50:14, FileCatalog.addFile, TargetSE StorageTest1, Successful, SequenceID 1
2015-08-19 12:38:20, FileCatalog.removeFile, Successful, SequenceID 2
2015-08-19 12:38:20, DataManager.removeFile, Successful, SequenceID 2
2015-08-19 12:38:20, StorageElement.removeFile, TargetSE StorageTest1, Successful, SequenceID 2
2015-08-19 12:38:20, FileCatalog.removeReplica, TargetSE StorageTest1, Successful, SequenceID 2

It is possible to get method calls for one method name, it is also possible to precise a date before, a date after or an operation Status.

For sequences, it is possible to get all sequences for an LFN, a user name, a group name, an host name, etc.

### CODE REVIEW

**Comment by atsareg** [comment]
Since this is your first PR, please pay attention to  not having a bad habbit of mixing code beautifications and real developments

**Comment by coberger** [comment]
Ok, thank you. I take notice for the next time.

File: DataManagementSystem/Agent/DataLoggingAgent.py [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+from DIRAC.Core.Base.AgentModule import AgentModule
+from DIRAC.DataManagementSystem.Client.test.mockDirac import ClientB
+
+# # agent's name
+AGENT_NAME = 'DataManagement/DataLoggingAgent'
+
+########################################################################
+class DataLoggingAgent( AgentModule ):
+  def initialize( self ):
```
Comment (phicharp) [inline DataManagementSystem/Agent/DataLoggingAgent.py]: If this agent is just for test, it should be in a test directory... Do you mean really an agent or should be a script?

Comment (coberger) [inline DataManagementSystem/Agent/DataLoggingAgent.py]: Sorry, I forgot to remove this file, it was to test something, it is not needed anymore.

**Comment by coberger** [comment]
Closed, move the PR on branch rel-v6r15

**Pull request closed without merging**